R/module_teal_lockfile.R

```diff
@@ -0,0 +1,195 @@
+#'      `implicit` type lockfile that uses `renv::dependencies()` to detect all R packages in the current project's
+#'      working directory.
+#'     - **`DESCRIPTION`-based lockfile**: To generate a lockfile based on a `DESCRIPTION` file in your working
+#'     directory, set `renv::settings$snapshot.type("explicit")`. The naming convention for `type` follows
+#'     `renv::snapshot()`. For the `"explicit"` type, refer to `renv::settings$package.dependency.fields()` for the
+#'     `DESCRIPTION` fields included in the lockfile.
+#'     - **Custom files-based lockfile**: To specify custom files as the basis for the lockfile, set
+#'     `renv::settings$snapshot.type("custom")` and configure the `renv.snapshot.filter` option.
+#'
+#' @section lockfile usage:
+#' After creating the lockfile, you can restore the application environment using `renv::restore()`.
+#'
+#' @seealso [renv::snapshot()], [renv::restore()].
+#'
+#' @return `NULL`
+#'
+#' @name module_teal_lockfile
+#' @rdname module_teal_lockfile
+#'
+#' @keywords internal
+NULL
+
+#' @rdname module_teal_lockfile
+ui_teal_lockfile <- function(id) {
+  if (!isTRUE(getOption("teal.renv.enable"))) {
+    return(NULL)
+  } else if (!.is_lockfile_deps_installed()) {
+    warning("Lockfile feature disabled. `mirai` and `renv` packages must be installed.")
+    return(NULL)
+  }
+
+  ns <- NS(id)
+  shiny::tagList(
+    tags$span("", id = ns("lockFileStatus")),
+    shinyjs::disabled(downloadLink(ns("lockFileLink"), "Download lockfile"))
+  )
+}
+
+#' @rdname module_teal_lockfile
+srv_teal_lockfile <- function(id) {
+  if (!isTRUE(getOption("teal.renv.enable")) || !.is_lockfile_deps_installed()) {
+    return(NULL)
+  }
+  moduleServer(id, function(input, output, session) {
+    logger::log_debug("Initialize srv_teal_lockfile.")
+    enable_lockfile_download <- function() {
+      shinyjs::html("lockFileStatus", "Application lockfile ready.")
+      shinyjs::hide("lockFileStatus", anim = TRUE)
+      shinyjs::enable("lockFileLink")
+    }
+    disable_lockfile_download <- function() {
+      warning("Lockfile creation failed.", call. = FALSE)
+      shinyjs::html("lockFileStatus", "Lockfile creation failed.")
+      shinyjs::disable("lockFileLink")
+    }
+
+    lockfile_path <- "teal_app.lock"
+    shiny::onStop(function() {
+      if (file.exists(lockfile_path) && !shiny::isRunning()) {
+        logger::log_debug("Removing lockfile after shutting down the app")
+        file.remove(lockfile_path)
+      }
+    })
+
+    # run renv::snapshot only once per app. Once calculated available for all users
+    if (file.exists(lockfile_path)) {
+      logger::log_debug("Lockfile has been already created - skipping automatic creation.")
+      enable_lockfile_download()
+      return(NULL)
+    }
+
+    # don't run renv::snapshot when option is set
+    user_lockfile <- getOption("teal.renv.lockfile", "")
```

Comment:
When I reading this again I started to think that the option is too complex - it consists of two technical terms: `renv` as well as `lockfile`. I think we can do simpler:
`teal.renv.lockfile` -> `teal.lockfile.path`
`teal.renv.enable` -> `teal.lockfile.enable`

The rationale is as follows:
- one technical term `lockfile` instead of two
- `renv` is irrelevant here - it should not patter how (which package) it is implemented

Please let me know your thoughts on that.

Reply:
Makes absolute sense 👍